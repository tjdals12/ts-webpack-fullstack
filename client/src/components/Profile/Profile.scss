.profile {
    width: 200px;
    height: 120px;
    border: 1px solid black;
}

.profile .name {
    font-size: 1.5rem;
}

.profile .age {
    font-size: 0.85rem;
}
