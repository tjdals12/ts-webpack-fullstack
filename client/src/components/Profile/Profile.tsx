import React from 'react';
import classNames from 'classnames/bind';
import styles from './Profile.scss';

const cx = classNames.bind(styles);

type ProfileProps = {
    name: string;
    age: number;
};

export default function Profile({
    name,
    age,
}: ProfileProps): React.ReactElement {
    return (
        <div className={cx('profile')}>
            <p className={cx('name')}>{name}</p>
            <p className={cx('age')}>{age}</p>
        </div>
    );
}
