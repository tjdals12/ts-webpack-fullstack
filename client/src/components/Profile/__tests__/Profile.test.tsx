import React from 'react';
import { render } from '@testing-library/react';
import Profile from '../Profile';

describe('<Profile />', () => {
    it('matches snapshot', () => {
        const utils = render(<Profile name="Seongmin" age={26} />);
        expect(utils.container).toMatchSnapshot();
    });

    it('has two text', () => {
        const utils = render(<Profile name="Seongmin" age={26} />);

        utils.getByText('Seongmin');
        utils.getByText('26');
    });
});
