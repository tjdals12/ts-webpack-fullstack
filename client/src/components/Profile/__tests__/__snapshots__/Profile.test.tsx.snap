// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div
    class="profile"
  >
    <p
      class="name"
    >
      Seongmin
    </p>
    <p
      class="age"
    >
      26
    </p>
  </div>
</div>
`;
